### PR TITLE
Remove isEnabled from the config model interface

### PR DESCRIPTION
**Pull request by bogdandrutu**
Fixes https://github.com/open-telemetry/opentelemetry-collector/issues/347

### CODE REVIEW

File: config/configmodels/configmodels.go [hunk 1]
```diff
@@ -40,22 +40,21 @@ type Config struct {
 	Service    Service
 }
 
-// NamedEntity is a configuration entity that has a name.
+// Component type as it is used in the config.
```
Comment (rghetia) [inline config/configmodels/configmodels.go:43]: doc should start with 'Type'.

File: config/configmodels/configmodels.go [hunk 2]
```diff
@@ -160,10 +150,6 @@ type Service struct {
 type ReceiverSettings struct {
 	TypeVal Type   `mapstructure:"-"`
 	NameVal string `mapstructure:"-"`
-	// Configures if the receiver is disabled and doesn't receive any data.
-	// The default value is false(meaning the receiver is enabled by default), and it is expected that receivers
-	// continue to use the default value of false.
-	Disabled bool `mapstructure:"disabled"`
 	// Configures the endpoint in the format 'address:port' for the receiver.
 	// The default value is set by the receiver populating the struct.
 	Endpoint string `mapstructure:"endpoint"`
```
Comment (rghetia) [inline config/configmodels/configmodels.go:155]: while you are at it, might as well fix the doc to start with 'Endpoint'.

File: config/configmodels/configmodels.go (hunk 1 above)
Comment (bogdandrutu) [reply config/configmodels/configmodels.go]: Done.

File: config/configmodels/configmodels.go (hunk 2 above)
Comment (bogdandrutu) [reply config/configmodels/configmodels.go:155]: Done.

**Pull request merged**